src/main/kotlin/org/jetbrains/plugins/bsp/magicmetamodel/impl/workspacemodel/impl/updaters/transformers/ModuleDetailsToGoModuleTransformer.kt

```diff
@@ -11,48 +11,51 @@ import org.jetbrains.plugins.bsp.magicmetamodel.impl.workspacemodel.GoModuleDepe
-            importPath = goBuildInfo.importPath ?: error("cos nie dziala"),
-            root = URI.create(inputEntity.target.baseDirectory).toPath(),
-            goDependencies = inputEntity.moduleDependencies.mapNotNull { targetsMap[it] }.map { buildTargetInfo ->
-                val buildTarget = projectDetails.targets.find { it.id == buildTargetInfo.id.toBsp4JTargetIdentifier() }
-                    ?: error("find nie działa")
-                val goBuildInfoo = extractGoBuildTarget(buildTarget) ?: error("extract nie działa")
-                GoModuleDependency(
-                    importPath = goBuildInfoo.importPath ?: error("nie ma importPatha"),
-                    root = URI.create(buildTarget.baseDirectory).toPath()
-                )
-            }
-        )
-    }
+    // TODO: replace importPath: String? in GoBuildTarget since Vgo modules require String
+    return GoModule(
+      module = toGenericModuleInfo(inputEntity),
+      importPath = goBuildInfo.importPath ?: "No importPath for GoModule",
+      root = URI.create(inputEntity.target.baseDirectory).toPath(),
+      goDependencies = toGoDependencies(inputEntity)
+    )
+  }
 
-    override fun toGenericModuleInfo(inputEntity: ModuleDetails): GenericModuleInfo {
-        val bspModuleDetails = BspModuleDetails(
-            target = inputEntity.target,
-            dependencySources = inputEntity.dependenciesSources,
-            type = type,
-            javacOptions = null,
-            pythonOptions = inputEntity.pythonOptions,
-            libraryDependencies = inputEntity.libraryDependencies,
-            moduleDependencies = inputEntity.moduleDependencies,
-            scalacOptions = inputEntity.scalacOptions,
-        )
+  override fun toGenericModuleInfo(inputEntity: ModuleDetails): GenericModuleInfo {
+    val bspModuleDetails = BspModuleDetails(
+      target = inputEntity.target,
+      dependencySources = inputEntity.dependenciesSources,
+      type = type,
+      javacOptions = null,
+      pythonOptions = inputEntity.pythonOptions,
+      libraryDependencies = inputEntity.libraryDependencies,
+      moduleDependencies = inputEntity.moduleDependencies,
+      scalacOptions = inputEntity.scalacOptions,
+    )
+    return bspModuleDetailsToModuleTransformer.transform(bspModuleDetails)
+  }
 
-        return bspModuleDetailsToModuleTransformer.transform(bspModuleDetails)
+  private fun toGoDependencies(inputEntity: ModuleDetails): List<GoModuleDependency>? {
+    val result = inputEntity.moduleDependencies.mapNotNull { targetsMap[it] }.mapNotNull { buildTargetInfo ->
+      val buildTarget = projectDetails.targets.find { it.id == buildTargetInfo.id.toBsp4JTargetIdentifier() }
+        ?: return@mapNotNull null
+      val dependencyGoBuildInfo = extractGoBuildTarget(buildTarget) ?: return@mapNotNull null
+        GoModuleDependency(
+          importPath = dependencyGoBuildInfo.importPath ?: "No importPath for GoModuleDependency",
```

Comment:
nitpick: I know that we have agreed to wait for checking what the default value should be, but we could declare this string as a constant somewhere

Reply:
defaultowa wartosc dla importpatha, czyli tam na gorze na przyklad

val NO_IMPORT_PATH = "No importPath for GoModule"